apiV2/midleware/auth.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable import/prefer-default-export */
+import Joi from '@hapi/joi';
+import jwt from 'jsonwebtoken';
+import { User } from '../models/users';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+
+class authMiddleware{
+  
+  static inputValidator(req, res, next) {
+    const authSchema = Joi.object().keys({
+      firstName: Joi.string().min(3).regex(/^[a-zA-Z\-]+$/).required(),
+      lastName: Joi.string().min(3).regex(/^[a-zA-Z\-]+$/).required(),
+      address: Joi.string().min(3).regex(/^[a-zA-Z0-9]+$/).required(),
+      password: Joi.string().regex(/^[a-zA-Z0-9]{3,30}$/).required(),
+      phoneNumber: Joi.string().regex(/^[a-zA-Z0-9]{10,30}$/).required(),
+      email: Joi.string().email({ minDomainSegments: 2 }).required(),
+      isAgent: Joi.required(),
+    });
+    const data = Joi.validate(req.body, authSchema);
+    if (typeof req.body.isAgent !== 'boolean') return res.status(400).send({ status: 400, error: 'isAgent should be a boolean' });
+    if (data.error) {
+      const resFomart = data.error.details[0].message.replace('"', '').split('"');
+      const gotElem = resFomart[0];
+      return res.status(400).send({ status: 400, error: `${gotElem} field  is invalid ` });
```

Comment:
@swaibat  don't you think you should have a helper function which handles your responses?  perhaps you just supply the status code, message and data as arguments.